### PR TITLE
✨ New Page - Migrated Live Page

### LINKED ISSUE #334: ✨ User Group Page - Live UG
~~⛔Do not proceed until the re-design mockup is added here.~~

Migrate https://www.ssw.com.au/ssw/Live/

Image

_Updated on Mar 7th 2023_
As per conversation with the TV team, this page needs a re-design to improve user experience.

```[tasklist]
### Tasks
- [x] Migrate page
- [x] Use SSW Google Account from Keeper to get YouTube API keys (maybe one of the accounts already has YouTube API provisioned)
- [x] Create a file in the services folder to fetch data from the YouTube API, then use in the `getStaticProps` function
- [x] Use [Incremental Static Regeneration](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) in a `live.tsx` page file to fetch latest videos in the UG playlist
- [x] Display "Upcoming Session Details" from SharePoint data (to also be fetched in getStaticProps using ISR) - same logic as in `netug/[[...filename]].tsx` and `events`
- [x] Display data under the "Upcoming Session Details" using `Event` in `events.tsx`
- [x] Update link of banner to point to the new page
- [x] Update Front Door
```

### PR DESCRIPTION
**Pull request by jeoffreyfischer**
- Description: The /live page needed to be migrated to the new Website. Each elements of the page is described in order in the Screenshots section below.

- Affected routes: /live

- Fixed #334 

- Done Video: https://www.youtube.com/watch?v=0Y5NF-NI1Lw

- Screenshots

live-page1
**Figure**: The page title (can be changed using Tina). The breadcrumb is shown below the title.

live-page2
**Figure**: The upcoming session section (Name can be changed in Tina - Content is loaded from SharePoint). The button at the bottom opens a new tab towards the [SSW TV channel](https://www.youtube.com/channel/UCBFgwtV9lIIhvoNh0xoQ7Pg).

live-page3
**Figure**: The past sessions recordings (Name can be changed in Tina - Content is loaded from YouTube using Google API). Only the last 6 videos are displayed. The button at the bottom opens a new tab towards the [YouTube Playlist](https://www.youtube.com/playlist?list=PLpiOR7CBNvlpmhfwQeIVhbqZKxV-do0wY).

### CODE REVIEW

File: .tina/collections/live.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      templates: [...Schemas.pageBlocks],
+    },
+    {
+      type: "string",
+      name: "section1",
+      label: "Next User Group",
+    },
+    {
+      type: "string",
+      name: "section2",
```
Comment (Harry-Ross) [inline .tina/collections/live.tsx]: Can you change the name of these?

File: .tina/config.tsx [hunk 2]
```diff
@@ -79,6 +80,7 @@ const config = defineStaticConfig({
       opportunitiesSchema,
       eventsIndexSchema,
       eventsSchema,
+      liveSchema,
```
Comment (Harry-Ross) [inline .tina/config.tsx]: Add this in alphabetical order (between industry and marketing)

File: components/button/utilityButton.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ type UtilityButtonProps = {
   noAnimate?: boolean;
   uncentered?: boolean;
   removeTopMargin?: boolean;
+  isTargetBlank?: boolean;
```
Comment (Harry-Ross) [inline components/button/utilityButton.tsx]: Can you rename this to something like `openInNewWindow` and can you also add this as a boolean to the Tina schema below.

File: components/live/header.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      <Container
+        className={classNames(
+          "flex-row  py-0 text-white md:flex text-center justify-center"
+        )}
+        size="custom"
+      >
+        <div className="flex max-w-3xl flex-col pb-10">
+          <div
+            className={classNames(
+              "flex flex-row items-center text-lg max-md:justify-center"
```
Comment (Harry-Ross) [inline components/live/header.tsx]:
```suggestion
              "flex flex-row items-center max-md:justify-center"
```

File: components/live/header.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            )}
+          ></div>
+          <div className="mb-2 pb-1 pt-3 text-5xl font-semibold child-h1:text-5xl child-p:text-xl descendant-strong:font-semibold descendant-strong:text-sswRed">
```
Comment (Harry-Ross) [inline components/live/header.tsx]: Should this be font-semibold for all elements?

File: components/usergroup/presenterLink.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  name: string;
+  profileLink?: string;
+};
+
+export const PresenterLink = ({
+  name,
+  profileLink = "/",
+}: PresenterLinkProps) => {
+  return <Link href={profileLink}>{name}</Link>;
+};
```
Comment (Harry-Ross) [inline components/usergroup/presenterLink.tsx]: Can you use `<Link>` instead of `<PresenterLink>` for simplicity?

File: content/live/index.mdx [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+section1: Upcoming Session Details
+section2: Recent Broadcasts
+youtubePlaylistId: PLpiOR7CBNvlpmhfwQeIVhbqZKxV-do0wY
+---
+
+# SSW **Live** Stream Channel
+
+User Group Live every 3rd Wednesday of the month at 6.30PM (AEDT UTC+11).
+
+Other special events - See dates on\_[Our Events](https://www.ssw.com.au/ssw/Events/)
```
Comment (Harry-Ross) [inline content/live/index.mdx]:
```suggestion
Other special events - See dates on\_[Our Events](https://www.ssw.com.au/events/)
```

File: components/live/header.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import classNames from "classnames";
+import { TinaMarkdown } from "tinacms/dist/rich-text";
+import { Container } from "../util/container";
+
+export const LiveHeader = (props) => {
```
Comment (Harry-Ross) [inline components/live/header.tsx]: Can you create a type for these props?

File: package.json [hunk 9]
```diff
@@ -55,6 +55,7 @@
     "dotenv": "^16.3.1",
     "eslint-config-next": "14.0.3",
     "formik": "^2.4.5",
+    "googleapis": "^128.0.0",
```
Comment (Harry-Ross) [inline package.json]: `yarn remove googleapis`

File: .tina/collections/live.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+export const liveSchema: Collection = {
+  label: "Live - Pages",
+  name: "live",
+  path: "content/live",
+  format: "mdx",
+  ui: {
+    router: ({ document }) => {
+      return `/live/${document._sys.filename}`;
+    },
+  },
```
Comment (Harry-Ross) [inline .tina/collections/live.tsx]:
```suggestion
  ui: {
    router: ({ document }) => {
      return "/live";
    },
    allowedActions: {
      create: false,
      delete: false,
    },
  },
```

File: pages/live.tsx [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+                  title={props.event.Title}
+                  theme="light"
+                />
+              ) : (
+                <VideoEmbed
+                  data={{
+                    url: "https://www.youtube.com/watch?v=coPZ75akNYA",
+                    videoWidth: "w-full",
+                    removeMargin: true,
+                  }}
```
Comment (Harry-Ross) [inline pages/live.tsx]: See what this looks like with a `VideoCard` for consistency

File: pages/live.tsx [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+            link="https://www.youtube.com/channel/UCBFgwtV9lIIhvoNh0xoQ7Pg"
+            buttonText={
+              <>
+                Visit SSW TV Channel on Youtube
+                <BsArrowRightCircle className="ml-1 inline" />
+              </>
+            }
+            noAnimate
+            isTargetBlank={true}
+          />
```
Comment (Harry-Ross) [inline pages/live.tsx]: Make link and the button text Tina fields.

File: pages/live.tsx [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+            .map((video, index) => (
+              <div key={index}>
+                <VideoCard {...video} theme="light" />
+              </div>
+            ))}
+        </div>
+        {props.playListVideosLinks.length > visibleVideos && (
+          <div className="flex justify-center">
+            <button
+              onClick={() => setVisibleVideos((prevCount) => prevCount + 9)}
```
Comment (Harry-Ross) [inline pages/live.tsx]: Can we make all instances of 9 a constant at the top of the file

File: pages/live.tsx [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+
+  let event = events[0];
+
+  if (!event) {
+    const pastEvents = await getEvents(
+      `$filter=fields/Enabled ne false and fields/EndDateTime lt '${currentDate}' and fields/CalendarType eq 'User Groups'&$orderby=fields/StartDateTime desc`
+    );
+
+    event = pastEvents[0];
+  }
```
Comment (Harry-Ross) [inline pages/live.tsx]: Can you abstract this to a file in `services/server` as its the same as in `pages/netug/[[..filename]].tsx`

File: pages/live.tsx [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+  return {
+    props: {
+      data: tinaProps.data,
+      query: tinaProps.query,
+      variables: tinaProps.variables,
+      playListVideosLinks,
+      event: event || null,
+      speaker: speaker || null,
+    },
+    revalidate: 10,
```
Comment (Harry-Ross) [inline pages/live.tsx]: Can you change this to an hour, and put a constant variable at the top of the file called `ISR_TIME` or something

File: pages/live.tsx [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+          <div>
+            {props.event?.Abstract && (
+              <div className="col-span-2">
+                <div className="whitespace-pre-wrap text-lg">
+                  {props.event?.Abstract}
+                </div>
+              </div>
+            )}
+          </div>
+          <div className="float-left">
```
Comment (Harry-Ross) [inline pages/live.tsx]:
```suggestion
          <div>
```

File: pages/live.tsx [hunk 17]
```diff
@@ -0,0 +1,203 @@
...
+                  name={props.speaker?.Title}
+                  profileLink={props.speaker?.PresenterProfileLink}
+                />
+              </div>
+              <br />
+            </>
+          )}
+        </div>
+        <div className="grid grid-cols-1 justify-center gap-8 lg:grid-cols-2">
+          <div>
```
Comment (Harry-Ross) [inline pages/live.tsx]:
```suggestion
          <div className="col-span-1">
```

File: pages/live.tsx [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+            {props.event?.Abstract && (
+              <div className="col-span-2">
```
Comment (Harry-Ross) [inline pages/live.tsx]:
```suggestion
              <div>
```

File: pages/live.tsx [hunk 19]
```diff
@@ -0,0 +1,203 @@
...
+                />
+              ) : (
+                <VideoEmbed
+                  data={{
+                    url: "https://www.youtube.com/watch?v=coPZ75akNYA",
+                    videoWidth: "w-full",
+                    removeMargin: true,
+                  }}
+                />
+              ))}
```
Comment (Harry-Ross) [inline pages/live.tsx]: Wrap in div with col-span-1

File: pages/live.tsx [hunk 20]
```diff
@@ -0,0 +1,203 @@
...
+    data: props.data,
+    query: props.query,
+    variables: props.variables,
+  });
+  const [visibleVideos, setVisibleVideos] = useState(9);
+
+  return (
+    <Layout>
+      <SEO seo={data.live.seo} />
+      <LiveHeader {...data.live} />
```
Comment (Harry-Ross) [inline pages/live.tsx]: Use explicit props once you've created the type for the props

**Review by Harry-Ross — CHANGES_REQUESTED** [review]

File: .tina/collections/live.tsx (hunk 1 above)
Comment (jeoffreyfischer) [reply .tina/collections/live.tsx]: Done

File: .tina/config.tsx (hunk 2 above)
Comment (jeoffreyfischer) [reply .tina/config.tsx]: Done

File: components/button/utilityButton.tsx (hunk 3 above)
Comment (jeoffreyfischer) [reply components/button/utilityButton.tsx]: Done

File: components/live/header.tsx (hunk 4 above)
Comment (jeoffreyfischer) [reply components/live/header.tsx]: Done

File: components/live/header.tsx (hunk 5 above)
Comment (jeoffreyfischer) [reply components/live/header.tsx]: Done - semibold is only applied to h1

File: components/usergroup/presenterLink.tsx (hunk 6 above)
Comment (jeoffreyfischer) [reply components/usergroup/presenterLink.tsx]: Done

File: content/live/index.mdx (hunk 7 above)
Comment (jeoffreyfischer) [reply content/live/index.mdx]: Done

File: components/live/header.tsx (hunk 8 above)
Comment (jeoffreyfischer) [reply components/live/header.tsx]: Done

File: package.json (hunk 9 above)
Comment (jeoffreyfischer) [reply package.json]: Done - 18 files removed 🥳

File: .tina/collections/live.tsx (hunk 10 above)
Comment (jeoffreyfischer) [reply .tina/collections/live.tsx]: Done

File: pages/live.tsx (hunk 11 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 12 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 13 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 15 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 16 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 17 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 18 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 19 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 20 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done

File: pages/live.tsx (hunk 14 above)
Comment (jeoffreyfischer) [reply pages/live.tsx]: Done - it is located in services | server | events.ts

File: content/live/index.mdx [hunk 21]
```diff
@@ -0,0 +1,19 @@
+---
+nextEvent: Upcoming Session Details
+sswTvButton:
+  name: Visit SSW TV Channel on Youtube
+  channelId: UCBFgwtV9lIIhvoNh0xoQ7Pg
+pastEvents: Recent Broadcasts
+youtubePlaylistButton:
+  name: Watch more on SSW TV Channel on Youtube
+  playlistId: PLpiOR7CBNvlpmhfwQeIVhbqZKxV-do0wY
+youtubePlaylistId: ''
```
Comment (Harry-Ross) [inline content/live/index.mdx]:
```suggestion
```

**Review by Harry-Ross — APPROVED** [review]
LGTM

File: content/live/index.mdx (hunk 21 above)
Comment (jeoffreyfischer) [reply content/live/index.mdx]: Done

**Review by wicksipedia — APPROVED** [review]
infra & actions - lgtm

**Pull request merged**